### PR TITLE
fix(sql): fix union all with all literals

### LINKED ISSUE #945: Union all does not work sometimes with SQL error
**Describe the bug**
```
select '2020-04-21', 1
union all
select '2020-04-22', 2
```
does not work. Fails with error

```
[17:21:15] select '2020-04-21', 1 union all select '2020-04-22', 2 ',', 'from' or 'over' expected
```
**Expected behavior**
Should work since 
```
select '2020-04-21' as a, 1 as b
union all
select '2020-04-22', 2
```
and 
```
select '2020-04-22', 2
```
are OK

**Environment (please complete the following information):**
 - Version: 5.0.6

### PR DESCRIPTION
**Pull request by puzpuzpuz**
Fixes #945

Note: `union` column and alias names won't be supported after this change, but that's should be OK since `union` is a reserved keyword in SQL.

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|1|1|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|1|1|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|1|1|100.00%|

**Comment by bluestreak01** [comment]
As a part of the package, fix one bug and find two more: what would happen with other set operations?

File: core/src/main/java/io/questdb/griffin/SqlParser.java [hunk 1]
```diff
@@ -1729,6 +1729,7 @@ private void validateLiteral(int pos, CharSequence tok) throws SqlException {
         columnAliasStop.add("from");
         columnAliasStop.add(",");
         columnAliasStop.add("over");
+        columnAliasStop.add("union");
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/griffin/SqlParser.java:1732]: there is also `except` and `intersect` operations that i suspect will have the same problem

**Review by bluestreak01 — CHANGES_REQUESTED** [review]

File: core/src/main/java/io/questdb/griffin/SqlParser.java (hunk 1 above)
Comment (puzpuzpuz) [reply core/src/main/java/io/questdb/griffin/SqlParser.java:1732]: Yes, that's what I'd also expect. Let me update the code and add some tests.

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|1|1|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|1|1|100.00%|

File: core/src/main/java/io/questdb/griffin/SqlParser.java (hunk 1 above)
Comment (puzpuzpuz) [reply core/src/main/java/io/questdb/griffin/SqlParser.java:1732]: Done in 9dba7f04ae7b56b2d31fd70111ccce0d170e077e

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|3|3|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|3|3|100.00%|

**Review by bluestreak01 — APPROVED** [review]

**Pull request merged**